cdap-data-fabric/src/main/java/co/cask/cdap/data2/registry/UsageRegistry.java

```diff
@@ -323,6 +325,26 @@ public Void apply(UsageDatasetIterable input) throws Exception {
   }
 
   /**
+   * Upgrades the UsageRegistry. In its current implementation it changed the type of usage.registry table in
+   * dataset.instances table from table to UsageDataset.
+   *
+   * @param datasetInstanceManager {@link DatasetInstanceManager} for the dataset instance meta data
+   */
+  public void upgrade(DatasetInstanceManager datasetInstanceManager) {
+    DatasetSpecification oldDatasetSpecification = datasetInstanceManager.get(USAGE_INSTANCE_ID);
+
+    if (!oldDatasetSpecification.getType().equals("UsageDataset")) {
```

Comment:
@chtyim : Is there a way to get this name programmatically when I don't have access to the dataset to do getName ?